misc-utils/fincore.c

```diff
@@ -357,39 +362,49 @@ static int fincore_fd (struct fincore_control *ctl,
  * Returns: <0 on error, 0 success, 1 ignore.
  */
 static int fincore_name(struct fincore_control *ctl,
-			struct fincore_state *st)
+			const char *filename,
+			const char *showname,
+			struct stat *statp)
 {
 	int fd;
 	int rc = 0;
-	struct stat sb;
+	struct stat sb, *stp = statp ?: &sb;
+	struct fincore_state st = { .name = filename };
 
```

Comment:
This is hard to read, there is `st` and `stp`, but both are for different types of structure. What about
```C
   struct stat _sb, *sb = statp  ?  : &_sb;
```

Reply:
and I would also suggest the same (_st, *st = &_st) for `fincore_state` so you do not need to change all code from `st->` to `st.`